mne/preprocessing/_csd.py

```diff
@@ -152,14 +189,17 @@ def compute_current_source_density(inst, sphere='auto', lambda2=1e-5,
     if not np.isfinite(pos).all() or np.isclose(pos, 0.).all(1).any():
         raise ValueError('Zero or infinite position found in chs')
     pos -= (x, y, z)
-
-    G = _calc_g(np.dot(pos, pos.T), stiffness=stiffness,
-                n_legendre_terms=n_legendre_terms)
-    H = _calc_h(np.dot(pos, pos.T), stiffness=stiffness,
-                n_legendre_terms=n_legendre_terms)
+    # Original code does not project onto the sphere, probably not good:
+    # pos /= radius
+    # from scipy.spatial.distance import squareform, pdist
+    # cos_dist = 1 - squareform(pdist(pos, 'sqeuclidean')) / 2.
+    # Project onto the sphere and do the distance (effectively)
+    pos /= np.linalg.norm(pos, axis=1, keepdims=True)
+    cos_dist = np.clip(np.dot(pos, pos.T), -1, 1)
+    del pos
+    G, H = _calc_G_H(n_legendre_terms, stiffness, cos_dist)
```

Comment:
The one thing I need to check is why `_calc_G_H` does not give the same result as `_calc_g` and `_calc_h`. That also seems to differ between this PR and `master`, but I haven't investigated exactly why.

Reply:
... oh wait, they actually match. I'll remove them in favor of `_calc_g` and `_calc_h`, since they're (much) simpler